qutebrowser/browser/commands.py

```diff
@@ -1737,6 +1739,12 @@ def fullscreen(self, leave=False):
 
         window = self._tabbed_browser.widget.window()
 
+        if enter:
+            if window.isFullScreen():
+                return
+            window.setWindowState(Qt.WindowFullScreen)
+            return
+
         if not window.isFullScreen():
             window.state_before_fullscreen = window.windowState()
         window.setWindowState(window.windowState() ^ Qt.WindowFullScreen)
```

Comment:
This properly preserves existing flags, while `window.setWindowState(Qt.WindowFullScreen)` above does not. Doing `window.setWindowState(window.windowState() | Qt.WindowFullScreen)` instead (with `--enter`) would probably be better. Then you can also get rid of the `window.isFullScreen()` check above, as setting the window state to the existing value doesn't do anything.

Reply:
implemented as suggested